File: CoWorkHub/CoWorkHub.Model/SpaceUnitResources.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class SpaceUnitResources
    {
        public int SpaceResourcesId { get; set; }
        public int SpaceUnitId { get; set; }
        public int ResourcesId { get; set; }
        public virtual Resource Resources { get; set; } = null!;
        public virtual SpaceUnit SpaceUnit { get; set; } = null!;
    }
}

File: CoWorkHub/UI/coworkhub_desktop/lib/models/space_unit_resources.dart
import 'package:json_annotation/json_annotation.dart';
import 'resource.dart';

part 'space_unit_resources.g.dart';

@JsonSerializable()
class SpaceUnitResources {
  int spaceResourcesId;
  int spaceUnitId;
  int resourcesId;
  Resource? resources;

  SpaceUnitResources({
    required this.spaceResourcesId,
    required this.spaceUnitId,
    required this.resourcesId,
    required this.resources,
  });

  factory SpaceUnitResources.fromJson(Map<String, dynamic> json) =>
      _$SpaceUnitResourcesFromJson(json);
  Map<String, dynamic> toJson() => _$SpaceUnitResourcesToJson(this);
}
